spec/services/notifications/remove_each_spec.rb

```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
+
+RSpec.describe Notifications::RemoveEach do
+  let(:user) { create(:user) }
+  let(:user2) { create(:user) }
+  let(:article) { create(:article, user_id: user.id) }
+  let(:comment) { create(:comment, user_id: user.id, commentable_id: article.id) }
+  let(:comment2) { create(:comment, user_id: user2.id, commentable_id: article.id) }
+  let(:mention) { create(:mention, user_id: user.id, mentionable_id: comment.id, mentionable_type: "Comment") }
+  let(:mention2) { create(:mention, user_id: user2.id, mentionable_id: comment2.id, mentionable_type: "Comment") }
+  let(:notifiable_collection) { [mention, mention2] }
+
+  before do
+    Notification.send_mention_notification_without_delay(mention)
```

Comment:
Regarding the previous comment, using `send_mention_notification_without_delay` could be ok since the `send_mention_notification` is not moved to ActiveJob and refactored yet. But still, if we use it now, we will need to modify this spec (so do more work) in the future. So I suggest creating the needed notifications explicitly here.

Reply:
I have removed the send_mention without delay and used notification + mention creation instead